docs/content/manual/advanced/addons/2764-vgpu.md

```diff
@@ -0,0 +1,152 @@
+---
+title: vGPU/SR-IOV GPU
+---
+
+
+* Related issues: [#1661](https://github.com/harvester/harvester/issues/2764) vGPU Support
+
+# Pre-requisite Enable PCI devices
+
+1. Create a harvester cluster in bare metal mode. Ensure one of the nodes has NIC separate from the management NIC
+1. Go to the management interface of the new cluster
+1. Go to Advanced -> PCI Devices
+1. Validate that the PCI devices aren't enabled
+1. Click the link to enable PCI devices
+1. Enable PCI devices in the linked addon page
+1. Wait for the status to change to `Deploy Successful`
+1. Navigate to the PCI devices page
+1. Validate that the PCI devices page is populated/populating with PCI devices
+
+# Pre-requisite Enable vGPU
+
+This can only be ran on a bare metal Harvester cluster that has an Nvidida card that support vGPU
+
+1. After the PCI devices are enabled navigate to the `nvidia-driver-toolkit` addon and enable it
+1. Wait for the status to change to `Deploy Successful`
+1. Edit the config for the `nvidia-driver-toolkit` from the addons page and set the driver location
+1. Navigate to SR-IOV GPU Devices
+1. Wait for the GPU to show up in the list
+1. Enable the GPU
+1. Wait for it to show as enabled and populate with the vGPU devices
+1. Navigate to vGPU devices
+1. Enable one of the vPGU devices and select a profile
+1. Validate that it now shows as enabled
+
+# Test Cases
+These tests were ran on Ubuntu focal KVM live images
+
+The setup for the VMs were as follows
+
+```sh
+wget https://developer.download.nvidia.com/compute/cuda/repos/wsl-ubuntu/x86_64/cuda-keyring_1.1-1_all.deb
+sudo dpkg -i cuda-keyring_1.1-1_all.deb
+sudo apt-get update
+sudo apt-get -y install cuda nvidia-cuda-toolkit build-essential
+
+git clone https://github.com/nvidia/cuda-samples
+cd cuda-samples/Samples/3_CUDA_Features/cudaTensorCoreGemm
+make
+./cudaTensorCoreGemm
+```
+
+## Add one vGPU on VM creation
+
+1. Create a VM and select the vGPU in vGPU devices
+1. After the VM is created run the pre-reqs as outlined above
+1. Run the grid installer for the vGPU driver on the VM
```

Comment:
Do we need explict command or webpage reference to install GRID vGPU driver package?
e.g.
```
sudo ./NVIDIA-Linux-x86_64-version-grid.run
```

Reply:
I agree that this is kind of unclear. The issue is that every different GPU will have a different URL, and you need to be an nvidia partner to log in and get the file. I don't have access to that. I'll add in a note that you need to get the KVM and grid files from Nvidia.

Reply:
I added a link in the pre-requisites that outlines which installers/drivers you need and where to see that Nvidia requires a partner program to download them.